### ISSUE TITLE
`from_str` rounding issues when overflow from too many digits

### ISSUE THREAD

**Issue by chris-cantor** [paupino/rust-decimal#453]
I am trying to add tests for the overflow behaviour of `from_str` and discovered a problem:

```rust
assert_eq!(
    Decimal::from_str("99999_99999_99999_99999_99999_99999.99999"),
    // it returns  Ok(100000_00000_00000_00000_00000_000) now
    Err(Error::from("Invalid decimal: overflow from too many digits"))
);
```

https://github.com/cantortrading/rust-decimal/runs/4657432688?check_suite_focus=true#step:9:87

It seems that the function is trying to round the input, such that 999,000 could be rounded to 1000,000 which is off by a lot.

What is the intended behaviour here? May be we should not attempt rounding before the decimal point?

**Comment by chris-cantor** [comment]
I have discovered another edge case: the decimal parsing lost accuracy when there are too many (i.e. > 28) leading 0s

```rust
assert_eq!(
    Decimal::from_str("00000_00000_00000_00000_00001.00001").unwrap(),
    Decimal::from_i128_with_scale(100001, 5) // it is now returning '1.0000'
);
```

https://github.com/cantortrading/rust-decimal/runs/4658904929?check_suite_focus=true#step:9:92

**Comment by paupino** [comment]
Fixed in #456
